Rna dnmp torsiondrives

chapincavender opened the pull request:
TorsionDrives for RNA dinucleoside monophosphates (DNMP), i.e. XpY 2-mers, on epsilon, zeta, alpha, beta, gamma, chi, 2' OH, 3' OH, and 5' OH with non-driven dihedrals constrained to A-form helix references values.

## New Submission Checklist

- [x] Created a new folder in the submissions directory containing the dataset
- [x] Added `README.md` describing the dataset see [here](https://github.com/openforcefield/qca-dataset-submission/tree/master/submissions/2020-03-26-OpenFF-Gen-2-Torsion-Set-6-supplemental-2) for examples
- [x] All files used to produce the dataset are included with a description
- [x] Dataset follows the QCSubmit schema defined for [Datasets](https://github.com/openforcefield/qcsubmit/blob/56680a7d3298b5d8962edcb840b0fdb34558c053/qcsubmit/datasets.py#L340), [OptimizationDatasets](https://github.com/openforcefield/qcsubmit/blob/56680a7d3298b5d8962edcb840b0fdb34558c053/qcsubmit/datasets.py#L1062) and [TorsionDriveDatasets](https://github.com/openforcefield/qcsubmit/blob/56680a7d3298b5d8962edcb840b0fdb34558c053/qcsubmit/datasets.py#L1225)
- [x] Dataset filename matches pattern `dataset*.json`; may feature a compression extension, such as `.bz2`
- [x] A PDF depicting the molecules is attached, in the case of torsiondrives this should include the highlighting of the central bond, this can be done automatically using [qcsubmit](https://github.com/openforcefield/qcsubmit/blob/56680a7d3298b5d8962edcb840b0fdb34558c053/qcsubmit/datasets.py#L854). 
- [x] QCSubmit validation passed
- [x] Made a new dataset entry in the mapping table in repository `README.md`
- [ ] Ready to submit!

ntBre reviewed (commented):
This looks reasonable to me. The only minor issue I noticed was the title in the submission README. I was also going to ask where the `initial-conformers` directory came from, but I saw that it was generated in the notebook. Do you think that needs to be included in the repository if the notebook can be used to generate it? Or are there parts of that directory not created by the notebook?

I'm also not sure I know enough about the science here to approve this conclusively. It might be better to ask Lily for a more thorough review on that front, but the general structure looks good to me.

chapincavender commented:
Thanks @ntBre! Good catch on the title in the README.

> I was also going to ask where the initial-conformers directory came from, but I saw that it was generated in the notebook. Do you think that needs to be included in the repository if the notebook can be used to generate it? Or are there parts of that directory not created by the notebook?

Everything in the `initial-conformers` directory is generated by the notebook, so I think it's reasonable to purge it from the repo. I had included similar files in previous datasets because the workflow involved a non-deterministic OpenMM minimization. But since I'm not doing that here I think those files can be reproduced exactly by running the notebook with the provided environment. Another option could be replacing the directory with a tarball.

lilyminium commented on submissions/2024-03-26-OpenFF-RNA-Dinucleoside-Monophosphate-TorsionDrives/README.md:
```diff
@@ -0,0 +1,53 @@
+# XtalPi Shared Fragments TorsiondriveDataset v1.0
```
Yes, please change this to the RNA title -- please also add a -v1.0 to the folder name to match the URLs given in the README and dataset.

lilyminium reviewed (changes requested):
Generally looks good! Please just fix up the README/folder names so the URLs match.

> But since I'm not doing that here I think those files can be reproduced exactly by running the notebook with the provided environment. Another option could be replacing the directory with a tarball.

I think this can differ across hardware too since initial conformers are generated randomly, albeit with a printed seed -- IMO just replacing the files with a tarball would be best here.

chapincavender commented:
Thanks @lilyminium, I fixed the dataset README and changed the folder name as requested. For the `initial-conformers` directory, I did `git reset --soft HEAD~2`, added the other files from the previous commits, then did `git push --force`. I think this will have removed the unwanted files from the repo history, but this is stretching the limits of my git knowledge. Please let me know if you want any other revisions.

lilyminium reviewed (approved):
LGTM -- thanks for the changes @chapincavender!

lilyminium commented:
@j-wags when you have a minute could you please start some workers for these torsiondrives?
